tests/test_salesforce_custom_fields_test.py

```diff
@@ -0,0 +1,43 @@
+"""
+Test that with no fields selected for a stream automatic fields and custom fields  are still replicated
+"""
+
+from tap_tester.base_suite_tests.all_fields_test import AllFieldsTest
+from sfbase import SFBaseTest
+
+
+class SFCustomFieldsTest(AllFieldsTest, SFBaseTest):
+    """Test that with no fields selected for a stream automatic fields and custom fields are still replicated
+    TODO
+    This test coveres only BULK api and just couple of streams. We have separate cards to cover these cases
+    TDL-23653: [tap-salesforce]: QA - Add all the streams for the all_fields test
+    TDL-23654: [tap-salesforce]: QA - Add all-fields testcase for REST API streams
+    """
+
+    salesforce_api = 'BULK'
+
+    @staticmethod
+    def name():
+        return "tt_sf_custom_fields_test"
+
+    def streams_to_test(self):
+        streams = {'Account', 'Contact'}
+        return streams
+
+    def test_custom_fields( self ):
+        expected_streams = self.selected_fields
```

Comment:
It does not appear that we are doing any filtering up front on field selection so even though this is the custom fields test I believe we will be selecting, and replicating all fields.

Selecting all fields but then only testing that we have selected and replicated the custom ones may result in a similar test but will be much less efficient than just selecting, replicating, and verifying the portion of streams and fields we want.  It could also possibly mask a field selection bug.

Reply:
I have overridden the methid streams_to_selected_fields to return a dict of stream and custom_fields for custom fields testcase and a dict of stream and non_custom_fields for non custom fields testcase